New package: ShiningLight.OpenSSL.Light version 3.3.1

Dragon1573 opened the pull request:
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] This resolves #157000

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/157003)

Dragon1573 committed:
New package: ShiningLight.OpenSSL.Light version 3.3.1

Signed-off-by: 是蓝兔哟～ <49941141+Dragon1573@users.noreply.github.com>

Dragon1573 commented:
> [!WARNING]
>
> Only `x86` and `x64` is tested in `amd64` _Windows Sandbox_ instance. There might be error properties of `amd64` installer. Appreciate if other volunteers & moderators can help validate it. 🙏🏼

> [!NOTE]
>
> This is a successor of `ShiningLight.OpenSSL` version `3.2.1`. I'd like to distinguish them as `ShiningLight.OpenSSL.Light` and `ShiningLight.OpenSSL.Dev`. `ShiningLight.OpenSSL` should be removed by another pull request.

stephengillie commented:
Verify: 
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/47747140-c156-4ec2-8851-490c65145a9b)
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/5e5b7378-8653-470b-8bfc-0ebcf3fafc4b)

Hi @Dragon1573,

The application shows version `3.3.1.4` while the Registry shows just `3.3.1` - this suggests the `PackageVersion` should be updated to `3.3.1.4` and match the application - so users will see this same version in both their application and WinGet. And a `DisplayVersion: 3.3.1` added so WinGet can still match with the Registry. 

What do you think?

Dragon1573 committed:
Apply suggesstion from code review.

- Change package version: 3.3.1 ==> 3.3.1.4
- Add "DisplayVersion" in ARP entries
